backend/src/repositories/CompanyRepository.ts

```diff
@@ -11,6 +12,12 @@ async function createCompany(companyInfo: AddCompanyRequestDto) {
     try {
         const companiesTable = await getCompanyTable();
 
+        const existingCompany = await findCompanyByName(companyInfo.name);
+
+        if (!!existingCompany) {
```

Comment:
Double negation is not a good practice since it makes the code not very readable. This is because our brain isn't that good at doing "not not X" instead we do something like this we change the variable from "existingCompany" to "isCompanyExisting" and negate that. For null or undefined we can simply have those as an if statement instead of double negating to evaluate to True when a company already exists.